src/mon/MgrMonitor.cc

```diff
@@ -856,6 +856,9 @@ bool MgrMonitor::preprocess_command(MonOpRequestRef op)
   string format;
   cmd_getval(g_ceph_context, cmdmap, "format", format, string("json-pretty"));
   boost::scoped_ptr<Formatter> f(Formatter::create(format));
+  if (!f) {
```

Comment:
in addition to removing the `if (!f) { f.reset(...); }` blocks in the branches below, i'd suggest apply a patch like
```diff
diff --git a/src/mon/MgrMonitor.cc b/src/mon/MgrMonitor.cc
index ff3539f72c..a706aab2f3 100644
--- a/src/mon/MgrMonitor.cc
+++ b/src/mon/MgrMonitor.cc
@@ -854,8 +854,10 @@ bool MgrMonitor::preprocess_command(MonOpRequestRef op)
   }

   string format;
-  cmd_getval(g_ceph_context, cmdmap, "format", format, string("json-pretty"));
-  boost::scoped_ptr<Formatter> f(Formatter::create(format));
+  cmd_getval(g_ceph_context, cmdmap, "format", format, {});
+  boost::scoped_ptr<Formatter> f(Formatter::create(format,
+                          "json-pretty",
+                          "json-pretty"));
```
instead of adding yet another `if (!f)` block here.